updated markov() to add tranpose keyword + default warning

murrayrm opened the pull request:
This PR partially addresses issue #395 by adding a `transpose` keyword to the `markov` function that allows transposed inputs.  In the current release, the `markov` function is not consistent with the normal time-series convention (see discussion in #395) so a warning message is generated if `transpose` is not set explicitly to `True`.

These files will need to be updated again in the 0.9.0 release so that the default for the `transpose` keyword changes from `True` to `False`.

bnavigator commented:
This PR itself LGTM. But are we sure that the algorithm actually works for real data? The data used in the test is worthless to determine any markov parameters. (https://github.com/python-control/python-control/issues/395#issuecomment-618418178)

murrayrm commented:
Good point on whether the calculation was right.  Looks like it wasn't...

In this latest push, I have:

* Updated the markov() algorithm to compute the correct coefficients (the old code is there for comparison, along with a new algorithm + reference).
* Reverted to the original `markov_test.py` (to make sure tests are backward compatible) and moved all of the "enhanced" testing into `markov_test_array.py`.
* Added a new unit test in `markov_test_array.py` to make sure the calculation is correct.
